dnsimple/dnsimple.go

```diff
@@ -51,7 +44,19 @@ type Client struct {
 
 // NewClient returns a new DNSimple API client.
 func NewClient(apiToken, email string) *Client {
-	c := &Client{ApiToken: apiToken, Email: email, HttpClient: &http.Client{}, BaseURL: baseURL, UserAgent: userAgent}
+	credentials := NewApiTokenCredentials(email, apiToken)
+	c := &Client{Credentials: credentials, HttpClient: &http.Client{}, BaseURL: baseURL, UserAgent: userAgent}
+	c.Contacts = &ContactsService{client: c}
+	c.Domains = &DomainsService{client: c}
+	c.Registrar = &RegistrarService{client: c}
+	c.Users = &UsersService{client: c}
+	return c
+}
+
+// NewAuthenticatedClient returns a new DNSimple API client  using the given
+// credentials.
+func NewAuthenticatedClient(credentials Credentials) *Client {
```

Comment:
Changing the existing `NewClient` will break any code that depends on this library, which is why I did not do it. Since we are sticking with this library supporting v1 in master, I am going to use option 1 instead, which is to have `NewClient` delegate to `NewAuthenticatedClient`. In v2 we can change the behavior, but that likely means moving the code to a brand new repo so we can use v1 and v2 in parallel.